main.tf

```diff
@@ -0,0 +1,45 @@
+data "aws_region" "current" {}
+
+data "template_file" "kubeconfig" {
+  template = "${file("${path.module}/templates/kubeconfig.tpl")}"
+
+  vars {
+    kubeconfig_name                   = "${local.kubeconfig_name}"
+    endpoint                          = "${var.cluster_endpoint}"
+    region                            = "${data.aws_region.current.name}"
+    cluster_auth_base64               = "${var.cluster_certificate_authority_data}"
+    aws_authenticator_command         = "${var.kubeconfig_aws_authenticator_command}"
+    aws_authenticator_command_args    = "${length(var.kubeconfig_aws_authenticator_command_args) > 0 ? "        - ${join("\n        - ", var.kubeconfig_aws_authenticator_command_args)}" : "        - ${join("\n        - ", formatlist("\"%s\"", list("token", "-i", var.cluster_id)))}"}"
```

Comment:
this is the thing i hate in terraform. If we can wrap around this then it is more readable.